src/common/crypto/mbedtls/driverwrapper.cpp

```diff
@@ -300,12 +300,18 @@ aos::RetWithError<KeyInfo> AosPsaAddKey(const aos::crypto::PrivateKeyItf& privKe
     for (psa_key_id_t keyID = MBEDTLS_PSA_KEY_ID_BUILTIN_MIN; keyID <= MBEDTLS_PSA_KEY_ID_BUILTIN_MAX; ++keyID) {
         bool found {};
 
+        psa_drv_slot_number_t maxSlotNumber = 0;
+
         for (auto& key : sBuiltinKeys) {
             if (key.mKeyID == keyID) {
                 found = true;
 
                 break;
             }
+
+            if (key.mSlotNumber > maxSlotNumber) {
```

Comment:
It means that the slot number is dynamic. If you remove one key, add another, and so on, the slot number will always increase. I strongly feel that slot number should be index in `sBuiltinKeys` array. E.g. `mSlotNumber` should not be part of `KeyDescription` but rather should be some flag as `mIntialized` for example. And when key is deleted this flag is reset when key added this flag is set and corresponding fields are updated.